app/src/main/java/com/example/aaa/pinkcalculator/Numbers.java

```diff
@@ -1,65 +1,87 @@
 package com.example.aaa.pinkcalculator;
 
+import android.widget.TextView;
+import static com.example.aaa.pinkcalculator.CommonConstants.MINUS;
+import static com.example.aaa.pinkcalculator.CommonConstants.POINT;
+import static com.example.aaa.pinkcalculator.CommonConstants.ZERO;
+
 public class Numbers {
+    private static final String MINUS_AND_NULL = "-0";
+
+    public static boolean canAddNumberToFirstNumberView(MainActivity mainActivity) {
+        return mainActivity.getFirstNumberView().getText().length() >= 0
+                && mainActivity.getOperationView().getText().length() == 0
+                && mainActivity.getSecondNumberView().getText().length() == 0
+                && mainActivity.getEquallyView().getText().length() == 0
+                && mainActivity.getResultNumber().getText().length() == 0;
+    }
+
+    public static boolean canAddNumberToSecondNumberView(MainActivity mainActivity) {
+        return mainActivity.getFirstNumberView().getText().length() > 0
+                && mainActivity.getOperationView().getText().length() > 0
+                && mainActivity.getSecondNumberView().getText().length() >= 0
+                && mainActivity.getEquallyView().getText().length() == 0
+                && mainActivity.getResultNumber().getText().length() == 0;
+    }
 
-    public static boolean canAddNumberToFirsNumberField(MainActivity mainActivity) {
-        return  mainActivity.getFieldForFirsNumber().getText().length() >= 0 &&
-                mainActivity.getFieldForOperation().getText().length() == 0 &&
-                mainActivity.getFieldForSecondNumber().getText().length() == 0 &&
-                mainActivity.getFieldForEqually().getText().length() == 0 &&
-                mainActivity.getFieldForResult().getText().length() == 0;
+    public static boolean canAddZeroToFirstNumberView(MainActivity mainActivity) {
```

Comment:
I don't see in your pull request where you use canAddZeroToFirstNumberView

Reply:
I use this method in MainActivity at the beginning of the method onClick

Reply:
Thanks. I founded it